backend/src/main/scala/com/softwaremill/adopttapir/infrastructure/CorrelationId.scala

```diff
@@ -1,43 +1,38 @@
 package com.softwaremill.adopttapir.infrastructure
 
+import cats.effect.std.Random
 import cats.effect.{IO, IOLocal}
 import sttp.capabilities.Effect
 import sttp.client3.*
 import sttp.monad.MonadError
 import sttp.tapir.server.interceptor.{EndpointInterceptor, RequestHandler, RequestInterceptor, Responder}
+import cats.syntax.all.*
 
-import scala.util.Random
+final class CorrelationId private (localCid: IOLocal[Option[String]], random: Random[IO]):
 
-object CorrelationId:
-  private val localCid =
-    import cats.effect.unsafe.implicits.global
```

Comment:
I refactored CorrelationId to avoid this `unsafeRunSync`. It is now initialized in `Main` and passed down as implicit.